src/core/dfn-contract.js

```diff
@@ -0,0 +1,61 @@
+export const name = "core/dfn-contract";
+
+export function run() {
+  addContractDefaults();
+  addDefinitionPointers();
+}
+
+function addContractDefaults() {
+  // Find all dfns that don't have a type and default them to "dfn".
+  /** @type NodeListOf<HTMLElement> */
+  const dfnsWithNoType = document.querySelectorAll(
+    "dfn:is([data-dfn-type=''],:not([data-dfn-type]))"
+  );
+  for (const dfn of dfnsWithNoType) {
+    dfn.dataset.dfnType = "dfn";
+  }
+
+  // Per "the contract", export all definitions, except where:
+  //  - Explicitly marked with data-noexport.
+  //  - The type is "dfn" and not explicitly marked for export (i.e., just a regular definition).
+  //  - definitions was included via (legacy) data-cite="foo#bar".
+  /** @type NodeListOf<HTMLElement> */
+  const exportableDfns = document.querySelectorAll(
+    "dfn:not([data-noexport], [data-export], [data-dfn-type='dfn'], [data-cite])"
+  );
+  for (const dfn of exportableDfns) {
+    dfn.dataset.export = "";
```

Comment:
I'm so confused... we were already doing this? I guess if it didn't break or do the wrong things it's ok? 🤷

Reply:
Not sure what you mean, but I just moved this content out from dfn.js so it can run before linters instead of after "plugins-done" event.